src/reducers/AuctionReducers.js

```diff
@@ -70,4 +171,106 @@ function auctions(state = defaultAuctionState, action) {
+                ubidId => bids[ubidId].bidderObj.uid === curBid.bidderObj.uid
+            );
+            if (matchBidderIndex !== -1) {
+                // console.log("There is a matching Bidder Bid at", bids[uniqueBids[matchBidderIndex]].bidAmount);
+                if (curBid.bidAmount > bids[uniqueBids[matchBidderIndex]].bidAmount) {
+                    // console.log("But the matching bid is less, so replace with current bid", curBid.bidAmount);
+                    uniqueBids[matchBidderIndex] = bidId;
+                } else {
+                    // console.log("And matching bid is more, so ignore current bid ", curBid.bidAmount)
+                }
+            } else {
+                
+                if (uniqueBids.length < 5) {
+                    // console.log("This bidder is unique and less than 5 unique bids so auto add");
+                    uniqueBids.push(bidId);
+                } else {
+                    // console.log("This bidder is unique but unique bids maxxed at ", uniqueBids.length);
+                    let lowestBid;
+                    
+                    uniqueBids.forEach( ubidId => {
+                        if (!lowestBid) {
+                            lowestBid = ubidId;
+                        } else {
+                            if( bids[ubidId].bidAmount < bids[lowestBid].bidAmount ) {
+                                lowestBid = ubidId;
+                            }
+                        }
+                    });
+                    
+                    if (bids[bidId].bidAmount > bids[lowestBid].bidAmount) {
+                        // console.log("but this bid is higher than lowest bid of "+bids[lowestBid].bidAmount+", so remove lowest and add this one")
+                        uniqueBids[uniqueBids.indexOf(lowestBid)] = bidId;
+                    } else {
+                        // console.log("lowest bid of "+bids[lowestBid].bidAmount+" is higher than this, ignore.")
+                    }
+                }
+            }
+        }
+    });
+            
+    uniqueBids.sort( (a,b) => {
+        if (bids[a].bidAmount < bids[b].bidAmount) {
+            return 1;
+        }
+        if (bids[a].bidAmount > bids[b].bidAmount) {
+            return -1;
+        }
+        return 0;
+    });
+    
+    uniqueBids.forEach( bidId => {
+        uniqueBidsObj[bidId] = bids[bidId];
+    });
+    
+    return uniqueBidsObj;
+}
+
+function processLoadedAuctionBids(auction) {
+    auction.bidTotal = 0;
+    auction.bids = filterUniqueBidders(auction.bids);
+    // if (auction.bids) {
+    //     let firstItem = auction.bids[Object.keys(auction.bids)[0]];
+    //     firstItem.checked = true;
+    //     auction.bidTotal += firstItem.bidAmount;
+    // } 
+    if (auction.winningBids) {
+        Object.keys(auction.winningBids).forEach( bidId => {
+            auction.bidTotal += auction.bids[bidId].bidAmount;
+            auction.bids[bidId].checked = true;
```

Comment:
I'm totally open to discussion with this but I found managing view model data that isn't held in the store was very difficult, as when I update the props on a stateless functional component it won't actually re-render the component, in Redux only when the state which is mapped to the component changes is the component re-rendered.